### DISCUSSION TITLE
Action not working with GitHub Enterprise 3.8

### DISCUSSION THREAD

**Discussion by JamzTheMan** [actions/create-github-app-token#49] [Q&A] [answered]
Followed the documentation here: https://docs.github.com/en/enterprise-server@3.8/apps/creating-github-apps/authenticating-with-a-github-app/making-authenticated-api-requests-with-a-github-app-in-a-github-actions-workflow

But I get the following error: `System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values. (Parameter ''using: node20' is not supported, use 'docker', 'node12' or 'node16' instead.')`

Looks like node20 is not yet available for Enterprise 3.8 runner (runner version 2.299.2)?

I tried rolling back to 1.0.5 but threw `Error: secretOrPrivateKey must be an asymmetric key when using RS256`
Changing v1.2.1 to use node16 also threw the same error?

Is there a minimum GitHub Enterprise version or runner version this action requires? Or is there another option?

**Comment by gr2m** [comment, marked as answer]
Hm sorry for that, we do depend on Node 20 I'm afraid, I'll make sure we revert back for GHES docs to recommend the previous action we recommended: https://github.com/tibdex/github-app-token

**Reply by JamzTheMan** [reply]
The downside of that is it's not an official "GitHub" action and we've restricted our GitHub Enterprise actions to not allow 3rd party actions for now. Also looks like v2.0.0 just moved to node 20 as well :(

**Reply by gr2m** [reply]
v2.0.0?

**Comment by JamzTheMan** [comment]
> v2.0.0?
Was referring to tibdex/github-app-token, everyone is going to node20 :)

**Reply by gr2m** [reply]
Got it, in that case try `tibdex/github-app-token@v1`

**Comment by skedwards88** [comment]
Thanks for bringing this up! The docs are now updated to recommend `tibdex/github-app-token` for GHES until GHES supports node 20. If your enterprise doesn't support third party actions, you can fork the `tibdex/github-app-token` action for your enterprise.
